src/vaults/factories/AsyncVaultFactory.sol

```diff
@@ -10,19 +10,23 @@ import {AsyncVault} from "src/vaults/AsyncVault.sol";
 import {IVaultFactory} from "src/vaults/interfaces/factories/IVaultFactory.sol";
 import {IPoolEscrowProvider} from "src/vaults/interfaces/factories/IPoolEscrowFactory.sol";
 import {IShareToken} from "src/vaults/interfaces/token/IShareToken.sol";
+import {IAsyncRedeemManager} from "src/vaults/interfaces/investments/IAsyncRedeemManager.sol";
 
 /// @title  ERC7540 Vault Factory
 /// @dev    Utility for deploying new vault contracts
 contract AsyncVaultFactory is Auth, IVaultFactory {
     address public immutable root;
-    address public immutable investmentManager;
+    IAsyncRedeemManager public immutable asyncRedeemManager;
```

Comment:
I think this should be `IAsyncRequests` instead because this is a fully async vault factory. No need to restrict the underlying investment manager.

Reply:
AFAIK, if it compiles, it's because internally the usage of the factory does not use more capabilities than `IAsyncRedeemManager` offer. Or am I missing something? It seems like the acctions from `IAsyncDepositManager` are only used from the PoolManager, not used by the vault itself.

Ideally , with time, would be cool to differentiate that logic

Reply:
> It seems like the actions from `IAsyncDepositManager` are only used from the `PoolManager`, not used by the vault itself. (EDIT: checked)

I disagree. The `AsyncVault` uses the `IAsyncDepositManager` via the `asyncManager()` in numerous codepaths: https://github.com/centrifuge/protocol-v3/blob/bd8cf05bc296c518b9811215e8d6880e9eff3ff7/src/vaults/AsyncVault.sol#L56

I can fix this with the below legacy vault investment manager interface and the merge of `IBaseInvestmentManager` and `IVaultManager`.

Reply:
Correct, there is a cast in `asyncManager()`, I missed that in my review. Sorry!

If we add `IAsyncRequests` as manager, we can then remove the `asyncManager()` method from the `AsyncVault`, at least for the internal usage.